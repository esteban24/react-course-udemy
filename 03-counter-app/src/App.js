import React from 'react';
import PropTypes from 'prop-types';
// import React, { Fragment } from 'react';

const App = ({ greeting, subtittle }) => {

    return (
        <>
            <h2>{ greeting }</h2>
            {/* <h1>{ message.name }, { message.age }</h1> */}
            {/* <pre>{ JSON.stringify(message, null, 3) }</pre> */}
            <p>{ subtittle }</p>
        </>
    );
}

/**
 * Forces greeting property to be passed as argument and checks wheter is string or not
 */
App.propTypes = {
    greeting: PropTypes.string.isRequired
}

App.defaultProps = {
    subtittle: 'First React Component'
}

export default App;