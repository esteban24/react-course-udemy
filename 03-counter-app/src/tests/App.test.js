import React from 'react';
import '@testing-library/jest-dom';
import { describe, expect, test } from '@jest/globals';
import { shallow } from 'enzyme';

// import { render } from '@testing-library/react'
import App from '../App';

describe('App.js test suite', () => {
    /**
     * Test without Enzyme, with extendeded-expect
     */
    // test('should show message "Hi, I\'m Goku"', () => {
    //     const greeting = "Hi, I\'m Goku";
    //     const { getByText } = render(<App greeting={ greeting }/>);
    //     expect(getByText(greeting)).toBeInTheDocument();
    // });

    test('should show <App /> properly', () => {
        const greeting = "Hi, I\'m Goku";
        const wrapper = shallow( <App greeting={ greeting }/>);

        expect(wrapper).toMatchSnapshot();
    })

    test('should return defined subtittle', () => {
        const greeting = "Hi, I\'m Goku";
        const subtittle = "Super sayan level 3"
        const wrapper = shallow( <App greeting={ greeting } subtittle={ subtittle }/>);

        const paragraphTest = wrapper.find('p').text();

        expect(paragraphTest).toBe(subtittle);
    })


});
