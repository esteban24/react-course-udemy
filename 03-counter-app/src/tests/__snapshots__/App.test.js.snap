// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js test suite should show <App /> properly 1`] = `
<Fragment>
  <h2>
    Hi, I'm Goku
  </h2>
  <p>
    First React Component
  </p>
</Fragment>
`;
